File: src/main/java/net/hectus/neobb/turn/person_game/warp/PTIceWarp.java
package net.hectus.neobb.turn.person_game.warp;

import com.marcpg.libpg.util.MinecraftTime;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.structure.PlacedStructure;
import net.hectus.neobb.util.Modifiers;
import org.bukkit.World;

public class PTIceWarp extends PWarpTurn {
    public PTIceWarp(World world) { super(world, "ice"); }
    public PTIceWarp(PlacedStructure data, World world, NeoPlayer player) { super(data, world, player, "ice"); }

    @Override
    public void apply() {
        super.apply();
        player.game.addModifier(Modifiers.G_PERSON_BLUE_ENTITIES);
        player.game.turnScheduler.runTaskTimer("icing", () -> player.game.players().forEach(p -> p.damage(1.0)), () -> true, 1);
        player.game.time(MinecraftTime.NOON);
    }

    @Override
    public int chance() {
        return 50;
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/person_game/warp/PTIceWarp.kt
package net.hectus.neobb.turn.person_game.warp

import com.marcpg.libpg.storing.Cord
import com.marcpg.libpg.util.MinecraftTime
import net.hectus.neobb.game.util.ScheduleID
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.structure.PlacedStructure
import net.hectus.neobb.structure.StaticStructure
import net.hectus.neobb.structure.StaticStructures
import net.hectus.neobb.util.Modifiers

class PTIceWarp(data: PlacedStructure?, cord: Cord, player: NeoPlayer?) : PWarpTurn(data, cord, "ice", player) {
    override val staticStructure: StaticStructure = StaticStructures.Person.Warp.ICE

    override val chance: Double = 50.0

    override fun apply() {
        player!!.game.addModifier(Modifiers.Game.Person.BLUE_ENTITIES)
        player.game.turnScheduler.runTaskTimer(ScheduleID.ICE, 1) {
            player.game.players.forEach { it.damage(1.0) }
        }
        player.game.time = MinecraftTime.NOON
    }
}
